tests/optimisticVerify.test.ts

```diff
@@ -0,0 +1,91 @@
+import { CurveType, Library, zkVerifySession } from '../src';
+import { walletPool } from './common/walletPool';
+import path from "path";
+import fs from "fs";
+
+jest.setTimeout(180000);
+
+describe('optimisticVerify functionality', () => {
+    let session: zkVerifySession;
+    let wallet: string;
+
+    const loadGroth16Data = () => {
+        const dataPath = path.join(__dirname, 'common/data', 'groth16_snarkjs_bls12381.json');
+        return JSON.parse(fs.readFileSync(dataPath, 'utf8'));
+    };
+
+    const createSessionAndInput = async (customWsUrl: string, publicSignals?: string[]) => {
+        const groth16Data = loadGroth16Data();
+        const { proof, publicSignals: defaultPublicSignals, vk } = groth16Data;
+
+        session = await zkVerifySession.start().Custom(customWsUrl).withAccount(wallet);
+
+        return {
+            session,
+            input: {
+                proofData: {
+                    proof,
+                    publicSignals: publicSignals || defaultPublicSignals,
+                    vk,
+                },
+            },
+        };
+    };
+
+    beforeEach(async () => {
+        wallet = await walletPool.acquireWallet();
+    });
+
+    afterEach(async () => {
+        if (session) await session.close();
+        if (wallet) await walletPool.releaseWallet(wallet);
+    });
+
+    it('should throw an error if optimisticVerify is called on a non-custom network', async () => {
+        session = await zkVerifySession.start().Testnet().withAccount(wallet);
+
+        const input = {
+            proofData: {
+                proof: {},
+                publicSignals: [],
+                vk: {},
+            },
+        };
+
+        await expect(
+            session.optimisticVerify()
+                .groth16(Library.snarkjs, CurveType.bn128)
+                .execute(input)
+        ).rejects.toThrowError('Optimistic verification is only supported on custom networks.');
+    });
+
+    it.skip('should succeed when called on a custom network with valid proof details', async () => {
```

Comment:
Are these meant to be skipped

Reply:
These are skipped because in order to pass they require a node that has exposed the unsafe methods (which includes dryRun)

Reply:
So you can only run these locally currently against your own node (which i did)

We can remove the skip if we ever get a dev node that exposes dryRun